### DISCUSSION TITLE
Resolve Fragment's activity in fragment scope?

### DISCUSSION THREAD

**Discussion by carterhudson** [InsertKoinIO/koin#1666] [Q&A]
I have some google sign in stuff that requires access to an Activity, but I want to abstract it away as `SaveEmailAuthCredentials` and hand it to my `ViewModel`. To avoid memory leaks, I'd like to use scopes. As such, my module looks like this:
```kotlin
val emailAuthModule = module {
    scope<EmailAuthFragment> {
        scoped {
            SaveEmailAuthCredentials(activity = androidContext().cast())
        }
        viewModel {
            EmailAuthViewModel(
                saveEmailAuthCreds = get<SaveEmailAuthCredentials>()
            )
        }
    }
}
```
My understanding is that `androidContext` should resolve the closest context to the scope, which would be the `EmailAuthFragment`'s Activity. However, when I use `androidContext()` here, it's retrieving the Application Context. Do I need to somehow nest my Fragment scope within an Activity scope in order to have access to the Activity?

**Comment by bpappin** [comment]
I have the same issue, I need to isolate the activity from the of the code, but i need th activity for the CredentialsManager, so I need it in the activity scope only.
